pkg/vault/test/authorizer/TimelockAuthorizerDelays.test.ts

```diff
@@ -0,0 +1,372 @@
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { advanceTime, currentTimestamp, DAY, MONTH } from '@balancer-labs/v2-helpers/src/time';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { range } from 'lodash';
+
+describe('TimelockAuthorizer delays', () => {
+  let authorizer: TimelockAuthorizer, vault: Contract;
+
+  let root: SignerWithAddress, other: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, root, other] = await ethers.getSigners();
+  });
+
+  const ACTION_1 = '0x0000000000000000000000000000000000000000000000000000000000000001';
+  const ACTION_2 = '0x0000000000000000000000000000000000000000000000000000000000000002';
+
+  const MINIMUM_EXECUTION_DELAY = 5 * DAY;
+
+  sharedBeforeEach('deploy authorizer', async () => {
+    let authorizerContract: Contract;
+
+    ({ instance: vault, authorizer: authorizerContract } = await Vault.create({
+      admin: root,
+      nextAdmin: ZERO_ADDRESS,
+    }));
+
+    authorizer = new TimelockAuthorizer(authorizerContract, root);
+  });
+
+  sharedBeforeEach('set delay to set authorizer', async () => {
+    const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+    // setAuthorizer must have a delay larger or equal than the one we intend to set - it is invalid to set any delays
+    // larger than setAuthorizer's.
+    // We set a very large setAuthorizer delay so that we have flexibility in choosing both previous and new delay
+    // values in the tests.
+    await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, DAY * 365, { from: root });
+  });
+
+  describe('scheduleDelayChange', () => {
+    const ACTION_DELAY = DAY;
+
+    function itSchedulesTheDelayChangeCorrectly(expectedExecutionDelay: number) {
+      it('schedules a delay change', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+
+        const { executed, data, where, executableAt } = await authorizer.getScheduledExecution(id);
+        expect(executed).to.be.false;
+        expect(data).to.be.equal(
+          authorizer.instance.interface.encodeFunctionData('setDelay', [ACTION_1, ACTION_DELAY])
+        );
+        expect(where).to.be.equal(authorizer.address);
+        expect(executableAt).to.equal((await currentTimestamp()).add(expectedExecutionDelay));
+      });
+
+      it('execution can be unprotected', async () => {
+        const id = await authorizer.scheduleDelayChange(ACTION_1, ACTION_DELAY, [], { from: root });
+        const execution = await authorizer.getScheduledExecution(id);
+
+        expect(execution.protected).to.be.false;
+      });
+
+      it('execution can be protected', async () => {
+        const executors = range(4).map(() => ethers.Wallet.createRandom().address);
```

Comment:
We have a wrapper for `ethers.Wallet.createRandom` in `constants.ts`.
Although I think that wrapper should use `.address` instead of `getAddress()` :sweat_smile: